### PR TITLE
Add Key Vault virtual machine extension examples and suggestions

### PR DESCRIPTION
**Pull request by mimckitt**
## Description

**Adding examples and suggestions to use the Azure Key Vault virtual machine extension when adding certificates to a VM instance. This method is more secure and we want to point customers to perform the action in this manner instead of manually adding the secret.**

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Compute/Compute/help/Add-AzVMSecret.md [hunk 1]
```diff
@@ -27,7 +27,24 @@ For more information about the cmdlets, see [Azure Key Vault Cmdlets](/powershel
 
 ## EXAMPLES
 
-### Example 1: Add a secret to a virtual machine
+### Example 1: Add a secret to a virtual machine using the Azure Key Vault virtul machine extension
+To install certificates on a virtual machine it is recommended to use the [Azure Key Vault virtual machine extension for Linux](https://docs.microsoft.com/azure/virtual-machines/extensions/key-vault-linux) or the [Azure Key Vault virtual machine extension for Windows](https://docs.microsoft.com/azure/virtual-machines/extensions/key-vault-windows).
+
+```
```
Comment (isra-fel) [inline src/Compute/Compute/help/Add-AzVMSecret.md:33]: Please use ```powershell

File: src/Compute/Compute/help/Add-AzVMSecret.md [hunk 2]
```diff
@@ -27,7 +27,24 @@ For more information about the cmdlets, see [Azure Key Vault Cmdlets](/powershel
...
+# Build settings
+    $settings = '{"secretsManagementSettings": 
+    { "pollingIntervalInS": "' + <pollingInterval> + 
+    '", "certificateStoreName": "' + <certStoreName> + 
+    '", "certificateStoreLocation": "' + <certStoreLoc> + 
+    '", "observedCertificates": ["' + <observedCert1> + '","' + <observedCert2> + '"] } }'
```
Comment (isra-fel) [inline src/Compute/Compute/help/Add-AzVMSecret.md:39]: You can define it PowerShell way then use `ConvertTo-Json`. It's more natural.

```suggestion
$settings = @{
    secretsManagementSettings = @{
        pollingIntervalInS       = "<pollingInterval>"
        certificateStoreName     = "<certStoreName>"
        certificateStoreLocation = "<certStoreLoc>"
        observedCertificates     = @("<observedCert1>", "<observedCert2>")
    } 
} | ConvertTo-Json
```

I'm not 100% sure if the result is as you expect so please give it a try

**Review by isra-fel — CHANGES_REQUESTED** [review]
Hi @mimckitt  please see my inline comments (they apply to both VM and VMSS). Thanks!

**Review by isra-fel — APPROVED** [review]
LGTM

**Comment by mimckitt** [comment]
@isra-fel fixed the issues. Let me know if I need to do anything else to get this merged.

**Comment by mimckitt** [comment]
@smotwani FYI

File: src/Compute/Compute/help/Add-AzVmssSecret.md [hunk 3]
```diff
@@ -27,8 +27,32 @@ For more information about the cmdlets, see [Azure Key Vault Cmdlets](/powershel
...
+PS C:\> $extPublisher = "Microsoft.Azure.KeyVault"
+PS C:\> $extType = "KeyVaultForLinux"
+# Add Extension to VMSS
+PS C:\> $vmss = Get-AzVmss -ResourceGroupName <ResourceGroupName> -VMScaleSetName <VmssName>
+PS C:\> Add-AzVmssExtension -VirtualMachineScaleSet $vmss  -Name $extName -Publisher $extPublisher -Type $extType -TypeHandlerVersion "2.0" -Setting $settings
+# Start the deployment
+PS C:\> Update-AzVmss -ResourceGroupName <ResourceGroupName> -VMScaleSetName <VmssName> -VirtualMachineScaleSet $vmss
 ```
+
+To install certificates on a virtual machine it is recommended to use the [Azure Key Vault virtual machine extension for Linux](https://docs.microsoft.com/azure/virtual-machines/extensions/key-vault-linux) or the [Azure Key Vault virtual machine extension for Windows](https://docs.microsoft.com/azure/virtual-machines/extensions/key-vault-windows).
```
Comment (smotwani) [inline src/Compute/Compute/help/Add-AzVmssSecret.md:52]: Shouldn't the above statement be added before the Example 1 that provides the steps to add a secret to the VMSS using the Azure Key Vault extension, also we should mention virtual machine scaleset vm or someting similar that points to the scaleset resource which is different than the stand alone vm resource.

File: src/Compute/Compute/help/Add-AzVMSecret.md [hunk 4]
```diff
@@ -27,8 +27,28 @@ For more information about the cmdlets, see [Azure Key Vault Cmdlets](/powershel
 
 ## EXAMPLES
 
-### Example 1: Add a secret to a virtual machine
+### Example 1: Add a secret to a virtual machine using the Azure Key Vault virtul machine extension
```
Comment (smotwani) [inline src/Compute/Compute/help/Add-AzVMSecret.md:30]: I believe the extension is called Azure Key Vault extension, also there is a type in virtual word.

Comment (mimckitt) [reply src/Compute/Compute/help/Add-AzVMSecret.md:30]: Will push and update.

File: src/Compute/Compute/help/Add-AzVmssSecret.md (hunk 3 above)
Comment (mimckitt) [reply src/Compute/Compute/help/Add-AzVmssSecret.md:52]: I basically just inserted the extension info into what was already there. I will revisit this and see how to make the order more clear.

**Pull request merged**